### DISCUSSION TITLE
Integration with Obsidian

### DISCUSSION THREAD

**Discussion by feedhopper** [super-productivity/super-productivity#2330] [Ideas]
I love this tool and would very much like to see an integration with Obsidian.
obsidian.md/
It is the best note-taking manager out there and uses the open markdown standard.

Ideal would be to have super-productivity as a plugin within Obsidian. This would solve the problem of having to access an external website in order to use and sync the tasks as Obsidian already offers sync to all platforms.

In addition to that, SP would greatly benefit from being exposed to a very big and growing global community.

If it is too much work to implement SP directly as a plugin, maybe you could consider a solution that makes easy sync between Obsidian and SP possible so that users can manage their tasks in Obsidian where the data for their work is contained.

**Comment by eclecticpassions** [comment]
I am an Obsidian user and I would love to integrate the two apps together too! Is it possible, Devs?

**Comment by dairymilkbatman** [comment]
me 3!

**Comment by bytrangle** [comment]
It's been a long time since I used Obsidian. It'd be more helpful If any of Obsidian-SP users can provide more details, or even diagrams for how SP should be integrated into Obsidian.

Your SP data is on your hard disk. Is it possible to sync it to Obsidian?

And even if it is, there're still a lot of questions:
- real time? certain time? on user's action
- one way or two way?

Are there any syncing plugins for Obsidian that I can refer to?

**Comment by Onsowon** [comment]
An integration would be great! With the depth that you have with, for example, Excalidraw.

I tried to open "https://app.super-productivity.com" with the "Custom Frames" plugin, but not even that.

SuperPro (cool workflow) on Obsidian (best possible environment) = Super Powers.

*Comment translated automatically, sorry if it has mistakes.

**Comment by bytrangle** [comment]
Upon further research, it seems like this integration, whatever form it takes,  will send Super Productivity data (tasks) to Obsidian, and that probably fall outside the scope of Super Productivity 🤔.

**Reply by kenlefeb** [reply]
If there were a way to configure SP to save/sync projects and tasks to a directory hierarchy of markdown files with YAML frontmatter (rather than syncing to a single JSON file, for instance), then you wouldn't really need to do anything for Obsidian users to feel like they were integrated! We could then point our SP sync to a directory that's inside our Obsidian vault, and both apps would be able to work with the same files.

**Comment by ash1khan** [comment]
I am an avid user of Obsidian and my workflow ties heavily with it. (From Project maintaining to future goals, From Book/Tutorial notes to Blog writing). When I discovered Super Productivity, I began to use it for several month with OpenProject Integration. As things get simplified and more straight forward, I totally moved totally from OpenProject (project tracking, git issues tracking, documentation, budget and task tracking) to Obsidian. My Initial thought to write a plugin for superProductivity that can pull only certain tags (e.g: #task) from obsidian and I can utilize pomodoro on those tasks. On completion of task it will inject the time to the exact location of that specific task on obsidian with checked thus simplifies more. But Js is not my taste at all. And angular is a messy heart attack for me. Seems like I am stuck here. Reverse plugin (Obsidian) can be written very easily where I can expose webhooks so that superProductivity can connect with obsidian. But superProductivity has no webhooks integration. And if I have to write obsidian plugin its better to implement the superProductivity style pomodoro into obisidian. One less app to open. Haha.

**Reply by mathisgauthey** [reply]
Hey there, if you're still up for such an obsidian plugin, I created a feature request on obsidian-tasks that might be of use : https://github.com/obsidian-tasks-group/obsidian-tasks/issues/2502

**Comment by OLLI-S** [comment]
I would love to have a Super Productivity integration in Obsidian.
SP could store the data in the Obsidian folder (or the online "Obsidian Sync") and I would have all my notes and tasks in Obsidian (and benefit from the Super Productivity features).
@johannesjo is this somehow possible or planned?

**Comment by johannesjo** [comment]
I think I found a good approach with the new plugin system. Stay tuned!

**Reply by aka-rider** [reply]
Something like that would be great. https://github.com/jamiebrynes7/obsidian-todoist-plugin

Bi-directional integration, basically.
This approach doesn’t impose a specific flow on users, so one could use Super Productivity to quickly add simple tasks on mobile (avoiding the heavy Obsidian vault loading), and plan medium to large projects in Obsidian—using, for example, Mermaid Gantt charts, documents from the knowledge base, PDF attachments.

The shared view matters, because in the end, execution time is shared.

**Comment by SoTHISIsFitness** [comment]
I just discovered SP, and my first thought was "omg integrate this with Obsidian!" 😂

It would definitely need to be two-way sync, and there's probably two main avenues we'd be looking at... either "each task is a file" or "tasks are all over the place, in many different files". Personally, I use the latter, and I think that could totally work provided we have projects be treated as a file, since those would naturally contain multiple tasks and other notes. Seems like a place to start, anyway.

**Reply by johannesjo** [reply]
The latter case sounds very complicated to deal with and a bit individual tbh. and I will go down an easier route with the official plugin, but with the new plugin system, you should be able to build whatever you need.

**Reply by aka-rider** [reply]
For the latter, Obsidian has a tasks plugin with its own API

**Comment by jakeingit** [comment]
I use Obsidian religiously every day.

The job to be done here is to leverage obsidian and the learning and notes we have -- not just bloat the amount of apps to get on top of.

What I do is run a agentic RAG agent that knows me and my tasks, ideas, etc. I feed it whatever structured data -- JSON, YAML, folders of notes. And it makes a vector store, or a knowledge graph. So when I setup an agentic agent, it knows how to answer questions, and can make decisions as good as me I give it specific information.

What I want to do is feed a sort of agentic AI thats in cahrge of a ((process)) or ((project)) and it can update data, or make API calls to other apps like n8n to automate workflows, then it gets some sort of feedback loop -- like a daily tracker.

For example, setting a goal, giving to AI, and it makes progress and reports, and shared problems and what its working on. Local AI agents take a lot of power and ram, and are 100x more slow, but full control with Pydantic AI and other frameworks.

I am not a programmer, but vibe coding using github repos and it works for me. Just chiming in, though. Not sure how I can help.

**Comment by joeljacques** [comment]
I would like something akin to the Citations plugin. It parses a .bib file and lets you search through the references and create new notes/text snippets based on that data and a given template. To be able to do this with SP tasks within seconds, through a hotkey and a couple symbols, would be great.

**Comment by rulrok** [comment]
Hi, since this app reminds me a lot todoist, I saw there is a todoist plugin to integrate with obsidian.
https://jamiebrynes7.github.io/obsidian-todoist-plugin/docs/overview
Maybe this can help guide on how the integration could go about

**Comment by sunsoy** [comment]
+1 SP and Obsidian integration would be a match made in heaven

**Comment by tandy-1000** [comment]
My use case would be; when I create a TODO section in obsidian notes, this should create a TODO in super-productivity

**Comment by mycochang** [comment]
Hey I used to use obsidian too, self-hosted w/ Working Copy iOS git hack 

Anyone else migrated/considering Anytype? The newest release is stable for self host and IMO is superior, and maintains the knowledge graph feature of obsidian. Which was my initial appeal. (Still love obsidian and it taught me how to write fluent .md)

**Comment by zitongcharliedeng** [comment]
+1 Maybe look at Obsidian's TaskNotes v4 plugin (TaskNotes.dev) since it has the "each task is a note" stuff and good extensibility, while using Obsidian bases too. Or some sort of markdown sync...

**Comment by Michaelknubben** [comment]
Really glad to see there is interest in this! I've been keeping an eye on Super Productivity, and the only thing stopping me from using it is that I use Obsidian for tasks

**Reply by johannesjo** [reply]
There already is a plugin you might make work for your use case.

**Reply by Michaelknubben** [reply]
Did you create one, or are you referring to this one? https://codeberg.org/sinenomine/sp-obsidian/
I've been hesitant to use that one as it's still beta (or at least... not on the official plugin repository), and I'd been holding off for your solution.

**Reply by johannesjo** [reply]
No. There is one pre-installed but not enabled. You need to use the desktop version.

**Reply by Michaelknubben** [reply]
Ah, I see, the sync.md addon? As I understand it from reading discussions on here, it only attaches to a single .md file? 
To better explain how I use tasks in Obsidian, let me show you a blurred copy: 
<img width="930" height="995" alt="image psd" src="https://github.com/user-attachments/assets/ec44ef9e-e4e9-4771-80fa-8968e1b8713e" />
Now, these don't come from a single .md, these are gathered from all of my notes. If I add a `-[  ]` to any note in Obsidian, it ends up in my tasks list, and I can either edit (toggle on/off) it in the original note, or in the overview.

edit: ... accidentally uploaded the unblurred copy, heh.

**Reply by Michaelknubben** [reply]
I apologise if this feels a little bit like this XKCD to someone not familiar with the workflow:
<img width="555" height="772" alt="image" src="https://github.com/user-attachments/assets/477b3a31-2a10-4d04-bf1e-c9943d5d412b" />
...Having Tasks embedded into my notes is what initially drew me to Obisdian, and it's fantastic! Syncing this with an external task app would give me the best of both worlds, as trying to shoehorn Pomodoro, project tracking, all that stuff... well it quickly gets away from the core of Obsidian for me, and SP does it so competently!

**Reply by FeralFlora** [reply]
> Did you create one, or are you referring to this one? https://codeberg.org/sinenomine/sp-obsidian/ I've been hesitant to use that one as it's still beta (or at least... not on the official plugin repository), and I'd been holding off for your solution.

I wish that one used block IDs as IDs, rather than markdown comments. It's the native way to reference blocks such as tasks.

**Reply by oswaltx** [reply]
I created a plugin which is in the plugin repository! Feedback and feature requests are welcome!
[https://github.com/oswaltx/obsidian-superproductivity-todo-sync](https://github.com/oswaltx/obsidian-superproductivity-todo-sync
)

**Comment by johannesjo** [comment]
Status update after re-reading the whole thread, since it has been open a while and there are now real options.

There are actually three different requests mixed in here, and they have very different answers:

**1. SP running inside Obsidian as a plugin.** Not happening, and I don't think it should. SP is a full Angular/Electron app; embedding it in a vault would mean maintaining a second app.

**2. SP tasks visible and editable inside Obsidian.** This is solved now, by two community plugins:

- [obsidian-superproductivity-todo-sync](https://github.com/oswaltx/obsidian-superproductivity-todo-sync) by @oswaltx, which is in the official Obsidian community plugin repo. It talks to SP's local REST API (no cloud, no third party server) and gives you a sidebar with your open tasks grouped by due date, project tabs, quick add with the same `@`/`#`/`+` syntax as SP, inline rename, and check off. Note it is a view onto SP's tasks, not a sync of your inline `- [ ]` checkboxes.
- [sp-obsidian](https://codeberg.org/sinenomine/sp-obsidian/) by sinenomine, which does 2 way sync from the SP side. Already listed in SP's plugin browser.

**3. Two way sync of `- [ ]` tasks scattered across every note in the vault.** This is what most of the recent comments are actually asking for (@tandy-1000, @Michaelknubben, @SoTHISIsFitness, @zitongcharliedeng), and it is the hard one. It needs a vault wide index, watching hundreds of files, per line conflict resolution, and an answer to "which SP project does this checkbox belong to?" that is genuinely different for every user. I am not going to build that into core. It belongs on the Obsidian side, where the vault index and the Tasks/Dataview APIs already exist.

**What already ships with SP today:** there is a bundled plugin called `sync.md`, pre installed but disabled. Enable it under Settings > Plugins on the desktop version. It does bidirectional sync between a markdown file and an SP project. Its limits, stated plainly:

- one file to one project (no multiple mappings yet)
- two levels of nesting, deeper levels become notes
- task IDs are stored as `<!-- sp:ID -->` HTML comments
- desktop only, because it needs Node file access

That last point matters for the "let Obsidian Sync handle all my platforms" idea (@kenlefeb, @OLLI-S): it won't, because sync.md doesn't run on mobile.

**What I'm open to improving in sync.md,** since it is the closest thing to a middle ground:

- multiple file to project mappings, so a vault with one note per project works. This is the main limitation people hit.
- recognising Obsidian block IDs (`^abc123`) as the task link where a line already has one, instead of only the HTML comment. @FeralFlora is right that comments are foreign to Obsidian.

What I'm not going to do is make markdown SP's storage format. It can't represent time tracking, repeat configs or the sync metadata, so it would be lossy in exactly the places that make SP useful.

If you're in group 3, try @oswaltx's plugin and give them feedback there. Concrete issues on a plugin that already exists will get you further than this thread will.

**Comment by Michaelknubben** [comment]
That's a good summary, except for the very final point: @oswaltx's plugin doesn't cover the needs of group 3.
If anyone ever does make something, please let me know!

**Reply by oswaltx** [reply]
What do you really need then. I am open to add features to my Plugin when you open issues

**Reply by Michaelknubben** [reply]
My tasks live inside of project notes inside of Obsidian. Currently I use the Tasks plugin to find these in my vault, based only on the - [ ] / - [ x ] standard, and it's a lovely way to work purely text-based for me. What I'd like is to be able to sync these with an external service, so I have access to them in a mobile app, and something on desktop to show timers, popups etc. 
Super Productivity seems perfect for this, so I'd love to integrate them

**Reply by Michaelknubben** [reply]
See here for an example: https://github.com/super-productivity/super-productivity/discussions/2330#discussioncomment-16851302

**Comment by giba0** [comment]
Hello Everyone

I just published my plugin that integrates Obsidian with Super Productivity (2-sync-way) I would love to receive your feedback:

[Obsidian Plugin ](https://community.obsidian.md/plugins/flowtask-super-productivity)

[Reddit post](https://www.reddit.com/r/superProductivity/comments/1wk1c95/flowtask_for_super_productivity_obsidian_and_sp/
)

I hope it’s useful for you!

**Reply by LittleSarina** [reply]
Brother i was literally just browsing some sort of solution to this
And then i found this comment :v
I'll check it
